### ISSUE TITLE
[Bug]: [nightly] Search failed reporting "no shard leaders available for channel..., Search 1 failed, reason cannot find segment 4"

### ISSUE THREAD

**Issue by binbinlv** [milvus-io/milvus#16666]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Environment

```markdown
- Milvus version:bc5e9ec (which is after the commit target to fixing this issue: b99b65c26e0967292171c83cc251939568de8d49)
- Deployment mode(standalone or cluster): cluster
- SDK version(e.g. pymilvus v2.0.0rc2):
- OS(Ubuntu or CentOS): 
- CPU/Memory: 
- GPU: 
- Others:
```

### Current Behavior

Search failed reporting "no shard leaders available for channel...":

(code=1, message=no shard leaders available for channel: by-dev-rootcoord-dml_195_432777434055573505v1, leaders: [6], err: fail to Search, QueryNode ID=6, reason=Search 1 failed, reason cannot find segment 432777434304610320 in QueryNode err %!w(<nil>))>

### Expected Behavior

Search successfully

### Steps To Reproduce

```markdown
Running nightly:
https://ci.milvus.io:18080/jenkins/blue/organizations/jenkins/milvus-nightly-ci/detail/master/492/pipeline/190/
(cluster)


1. Milvus logs:
https://ci.milvus.io:18080/jenkins/blue/organizations/jenkins/milvus-nightly-ci/detail/master/492/artifacts
name: artifacts-milvus-distributed-pulsar-master-492-pymilvus-e2e-logs.tar.gz

2. collection name: search_collection_7sY81VLw

3. Failed timeline:
[2022-04-25T19:28:25.487Z] [gw3] [ 63%] FAILED testcases/test_search_20.py::TestCollectionSearch::test_search_collection_after_release_load[5000-500-8-True-True] 

4. Failed test cases: test_search_collection_after_release_load

@pytest.mark.tags(CaseLabel.L1)
    def test_search_collection_after_release_load(self, nb, nq, dim, auto_id, _async):
        """
        target: search the pre-released collection after load
        method: 1. create collection
                2. release collection
                3. load collection
                4. search the pre-released collection
        expected: search successfully
        """
        # 1. initialize without data
        collection_w, _, _, insert_ids, time_stamp = self.init_collection_general(prefix, True, nb,
                                                                                  1, auto_id=auto_id,
                                                                                  dim=dim)[0:5]
        # 2. release collection
        log.info("test_search_collection_after_release_load: releasing collection %s" % collection_w.name)
        collection_w.release()
        log.info("test_search_collection_after_release_load: released collection %s" % collection_w.name)
        # 3. Search the pre-released collection after load
        log.info("test_search_collection_after_release_load: loading collection %s" % collection_w.name)
        collection_w.load()
        log.info("test_search_collection_after_release_load: searching after load")
        vectors = [[random.random() for _ in range(dim)] for _ in range(nq)]
        collection_w.search(vectors[:nq], default_search_field, default_search_params,
                            default_limit, default_search_exp, _async=_async,
                            travel_timestamp=time_stamp,
                            check_task=CheckTasks.check_search_results,
                            check_items={"nq": nq,
                                         "ids": insert_ids,
                                         "limit": default_limit,
                                         "_async": _async})



5. test logs:
https://ci.milvus.io:18080/jenkins/blue/rest/organizations/jenkins/pipelines/milvus-nightly-ci/branches/master/runs/492/nodes/190/steps/211/log/?start=0
```

**Comment by binbinlv** [comment]
@congqixia 
it seems the pr #16620  did not fix this issue, please investigate it again, thanks.

**Comment by congqixia** [comment]
Segment 432777434304610320 is released before search. This is not an expected behavior in query node. The root cause was that segment control cannot block segment removing operation. Fixing PR is on the way.

**Comment by congqixia** [comment]
@binbinlv after #16695 this problem did not reproduce in last 2 nightly.(Run 495 & 496). Could you please verify in next runs?
/unassign
/assign @binbinlv

**Comment by binbinlv** [comment]
OK，I will verify it in the following nightly tests, thanks.

**Comment by binbinlv** [comment]
After fix #16695 merged, this issue has not been reproduced for several days, so close it.
